### PR TITLE
Refactors o-video to follow the spec and to remove all unused code

### PR DESCRIPTION
**Pull request by AlbertoElias**
Also refactors tests and updates readme

Test coverage isn't complete, I need to prioritise getting this out and I still need to fix a few visual bugs, but I think it should be good enough

### CODE REVIEW

File: src/ads.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+			case google.ima.AdEvent.Type.STARTED:
+				// This event indicates the ad has started - the video player
+				// can adjust the UI, for example display a pause button and
+				// remaining time.
+				if (ad.isLinear()) {
+					// For a linear ad, a timer can be started to poll for
+					// the remaining time.
+					intervalTimer = setInterval(() => {
+						// Currently not used
+						// const remainingTime = this.adsManager.getRemainingTime();
```
Comment (rowanmanning) [inline src/ads.js:167]: Does this case need to be in here if we're not using it?

File: test/video.test.js [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+		});
+
+		it('should return 0 if duration is not set', () => {
+			video.getProgress().should.equal(0);
+		});
+
+		it('should return the progress of the video as a percentage', () => {
+			video.videoEl.duration = 100;
+			video.videoEl.currentTime = 30;
+			video.getProgress().should.equal(30);
```
Comment (rowanmanning) [inline test/video.test.js]: Not a big thing at all, but I'd be tempted to set these in a way that `getProgress()` doesn't return the same value as `currentTime`. E.g.

``` js
video.videoEl.duration = 200;
video.videoEl.currentTime = 50;
video.getProgress().should.equal(25);
```

**Comment by rowanmanning** [comment]
Looks great 👍 tests are very readable

File: src/ads.js (hunk 1 above)
Comment (AlbertoElias) [inline src/ads.js:167]: It's something that's more like a todo, I'll update the comment

File: test/video.test.js (hunk 2 above)
Comment (AlbertoElias) [inline test/video.test.js]: Yup!

**Pull request merged**